shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java

```diff
@@ -192,6 +207,101 @@ public String downloadDicomMetadataForURL(final URL url) throws IOException, Mes
+        if (examination != null && examination.getDatasetAcquisitions() != null && !examination.getDatasetAcquisitions().isEmpty()
+                && examination.getDatasetAcquisitions().get(0) != null && examination.getDatasetAcquisitions().get(0).getDatasets() != null
+                && !examination.getDatasetAcquisitions().get(0).getDatasets().isEmpty() 
+                && examination.getDatasetAcquisitions().get(0).getDatasets().get(0) != null) {
+            List<URL> urls = new ArrayList<>();
+            try {
+                DatasetUtils.getDatasetFilePathURLs(examination.getDatasetAcquisitions().get(0).getDatasets().get(0), urls, DatasetExpressionFormat.DICOM);
+                if  (!urls.isEmpty()) {
+                    return urls.get(0).toString();
+                }
+            } catch (MalformedURLException e) {
+                return null;
+            }
+        }
+        return null;
+    }
+    
+    public Attributes getDicomAttributesForStudy(Study study) {
+        long ts = new Date().getTime();
+        Examination exam = getFirstIfExist(study.getExaminations());
+        if (exam == null) return null;
+        Attributes result = getDicomAttributesForExamination(exam);
+        LOG.error("get DICOM attributes for study " + study.getId() + " : " + (new Date().getTime() - ts) + " ms");
+        return result;
+    }
+    
+
+    public Attributes getDicomAttributesForExamination(Examination examination) {
+        long ts = new Date().getTime();
+        DatasetAcquisition acquisition = getFirstIfExist(examination.getDatasetAcquisitions());
+        if (acquisition == null) return null;
+        Attributes result = getDicomAttributesForAcquisition(acquisition);
+        LOG.error("get DICOM attributes for acquisition " + acquisition.getId() + " : " + (new Date().getTime() - ts) + " ms");
+        return result;
+    }
+
+    public Attributes getDicomAttributesForAcquisition(DatasetAcquisition acquisition) {
+        long ts = new Date().getTime();
+        Dataset ds = getFirstIfExist(acquisition.getDatasets());
+        if (ds == null) return null;
+        Attributes result = getDicomAttributesForDataset(ds);
+        LOG.error("get DICOM attributes for dataset " + ds.getId() + " : " + (new Date().getTime() - ts) + " ms");
+        return result;
+    }
+    
+    public String getDicomJson(Examination examination) throws IOException {
+        String urlStr = getExaminationFirstDatasetUrl(examination);
+        if (urlStr != null) {
+            if (urlStr.contains(WADO_REQUEST_STUDY_WADO_URI)) urlStr = wadoURItoWadoRS(urlStr);
+            urlStr = urlStr.split(CONTENT_TYPE)[0];
+            urlStr = urlStr.split("/series/")[0];
+            urlStr = urlStr.concat("/metadata/");
+            String json = downloadMetadataFromPACS(urlStr);
+            // transform from flat to tree
+            try {
+                System.out.println(json);
```

Comment:
Hi Julien, it would be great to remove this System.out.println please. Thank you!